azurerm/resource_arm_backup_policy_file_share.go

```diff
@@ -0,0 +1,379 @@
+			TimeZone:             utils.String(d.Get("timezone").(string)),
+			BackupManagementType: backup.BackupManagementTypeAzureStorage,
+			WorkLoadType:         backup.WorkloadTypeAzureFileShare,
+			SchedulePolicy:       expandArmBackupProtectionPolicyFileShareSchedule(d, times),
+			RetentionPolicy: &backup.LongTermRetentionPolicy{ //SimpleRetentionPolicy only has duration property ¯\_(ツ)_/¯
+				RetentionPolicyType: backup.RetentionPolicyTypeLongTermRetentionPolicy,
+				DailySchedule:       expandArmBackupProtectionPolicyFileShareRetentionDaily(d, times),
+			},
+		},
+	}
+	if _, err = client.CreateOrUpdate(ctx, vaultName, resourceGroup, policyName, policy); err != nil {
+		return fmt.Errorf("Error creating/updating Recovery Service Protection Policy %q (Resource Group %q): %+v", policyName, resourceGroup, err)
+	}
+
+	resp, err := resourceArmBackupProtectionPolicyFileShareWaitForUpdate(ctx, client, vaultName, resourceGroup, policyName, d)
+	if err != nil {
+		return err
+	}
+
+	id := strings.Replace(*resp.ID, "Subscriptions", "subscriptions", 1)
+	d.SetId(id)
+
+	return resourceArmBackupProtectionPolicyFileShareRead(d, meta)
+}
+
+func resourceArmBackupProtectionPolicyFileShareRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).RecoveryServices.ProtectionPoliciesClient
+	ctx, cancel := timeouts.ForRead(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	policyName := id.Path["backupPolicies"]
+	vaultName := id.Path["vaults"]
+	resourceGroup := id.ResourceGroup
+
+	log.Printf("[DEBUG] Reading Recovery Service Protection Policy %q (resource group %q)", policyName, resourceGroup)
+
+	resp, err := client.Get(ctx, vaultName, resourceGroup, policyName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Recovery Service Protection Policy %q (Resource Group %q): %+v", policyName, resourceGroup, err)
+	}
+
+	d.Set("name", policyName)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("recovery_vault_name", vaultName)
+
+	if properties, ok := resp.Properties.AsAzureFileShareProtectionPolicy(); ok && properties != nil {
+		d.Set("timezone", properties.TimeZone)
+
+		if schedule, ok := properties.SchedulePolicy.AsSimpleSchedulePolicy(); ok && schedule != nil {
+			if err := d.Set("backup", flattenArmBackupProtectionPolicyFileShareSchedule(schedule)); err != nil {
+				return fmt.Errorf("Error setting `backup`: %+v", err)
+			}
+		}
+
+		if retention, ok := properties.RetentionPolicy.AsLongTermRetentionPolicy(); ok && retention != nil {
+			if s := retention.DailySchedule; s != nil {
+				if err := d.Set("retention_daily", flattenArmBackupProtectionPolicyFileShareRetentionDaily(s)); err != nil {
+					return fmt.Errorf("Error setting `retention_daily`: %+v", err)
+				}
+			} else {
+				d.Set("retention_daily", nil)
+			}
```

Comment:
I don't think this is required?

Reply:
You're right. This is left over from the VM backup policy which supports multiple types of retention settings that are configured individually. One or all of `retention_daily`, `retention_weekly`, `retention_monthly`, and `retention_yearly` can be set. I can  remove it for now, but it will probably need to be added back if Azure Backup eventually supports different retentions